### PR TITLE
Feature/analysis export

### PR DESCRIPTION
**Pull request by k9845**
Addresses 
- Analysis Export(https://github.com/the-deep/server/issues/1091)

## Changes
- Add mutation for analysis export

Mention related users here if any.

## This PR doesn't introduce any:

- [ ] temporary files, auto-generated files or secret keys
- [ ] n+1 queries
- [ ] flake8 issues
- [ ] `print`
- [ ] typos
- [ ] unwanted comments

## This PR contains valid:

- [ ] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/export/analyses/excel_exporter.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            'Statement',
+            'Entry ID',
+            'Entry',
+            'Entry Link',
+            'Source Link'
+        ]
+
+    def add_analytical_statement_entries(self, analytical_statement_entries):
+        self.analysis_sheet.append([self.titles])
+        for analytical_statement_entry in analytical_statement_entries:
```
Comment (thenav56) [inline apps/export/analyses/excel_exporter.py]: Let's use iterator for analytical_statement_entries

File: apps/export/analyses/excel_exporter.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            url = f'{settings.HTTP_PROTOCOL}://{settings.DEEPER_FRONTEND_HOST}/permalink/projects/'
```
Comment (thenav56) [inline apps/export/analyses/excel_exporter.py]: Let's create a function for this. Maybe define a file like deep/premalinks.py and define permalink patterns and functions there.

File: apps/export/serializers.py [hunk 3]
```diff
@@ -290,6 +294,14 @@ def validate(self, data):
         data_type = data['type']
         export_type = data['export_type']
         _format = data['format']
+        project = self.project
+        analysis = data.get('analysis')
+        if analysis:
+            analysis_object = get_object_or_404(Analysis, id=analysis.id)
+            if analysis_object.project != project:
```
Comment (thenav56) [inline apps/export/serializers.py]:
```suggestion
            if analysis_object.project != self.project:
```

File: apps/export/serializers.py [hunk 4]
```diff
@@ -290,6 +294,14 @@ def validate(self, data):
...
+                raise serializers.ValidationError(
+                    f'Analysis project {analysis.project_id} does not current project {project.id}'
+                )
```
Comment (thenav56) [inline apps/export/serializers.py]: We can move this to validate_analysis

File: apps/export/serializers.py [hunk 5]
```diff
@@ -316,6 +328,7 @@ def create(self, data):
                 'report_structure',
             ) if key in data
         }
+        data['analysis'] = data.get('analysis')
```
Comment (thenav56) [inline apps/export/serializers.py]:
```suggestion
```

File: apps/export/tasks/tasks_analyses.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+
+def export_analyses(export):
+    export_type = export.export_type
+    analysis = export.analysis
+
+    analytical_statement_entries = AnalyticalStatementEntry.objects.filter(
+        analytical_statement__analysis_pillar__analysis=analysis
+    )
+    print(analytical_statement_entries, "*******************")
```
Comment (thenav56) [inline apps/export/tasks/tasks_analyses.py]:
```suggestion
```

File: apps/export/tasks/tasks_analyses.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    if export_type == Export.ExportType.EXCEL:
+        export_data = ExcelExporter(analytical_statement_entries)\
+            .add_analytical_statement_entries(analytical_statement_entries)\
+            .export()
+    else:
+        raise Exception(
+            '(Entries Export) Unkown Export Type Provided: {export_type} for Export: {export.id}'
```
Comment (thenav56) [inline apps/export/tasks/tasks_analyses.py]:
```suggestion
           f'(Entries Export) Unkown Export Type Provided: {export_type} for Export: {export.id}'
```

File: apps/export/tests/test_mutations.py [hunk 8]
```diff
@@ -256,6 +263,44 @@ def _query_check(minput, **kwargs):
...
+        self.force_login(self.non_member_user)
+        _query_check(minput, assert_for_error=True)
+
+        # --- member user
+        self.force_login(self.member_user)
+
+        response = _query_check(minput, okay=False)['data']
+        self.assertNotEqual(response['project']['exportCreate']['result'], None, response)
+        self.assertEqual(response['project']['exportCreate']['result']['analysis']['title'], analysis1.title)
+
```
Comment (thenav56) [inline apps/export/tests/test_mutations.py:303]: Where is test for file check?

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: apps/export/tests/test_mutations.py (hunk 8 above)
Comment (k9845) [reply apps/export/tests/test_mutations.py:303]: Added `TODO`

File: apps/export/analyses/excel_exporter.py [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+                analytical_statement_entry.analytical_statement.analysis_pillar_id,
+                analytical_statement_entry.analytical_statement.analysis_pillar.title,
+                analytical_statement_entry.analytical_statement.analysis_pillar.assignee.get_display_name(),
+                analytical_statement_entry.analytical_statement_id,
+                analytical_statement_entry.analytical_statement.statement,
+                analytical_statement_entry.entry_id,
+                analytical_statement_entry.entry.excerpt,
+                url,
+                analytical_statement_entry.entry.lead.url,
+
```
Comment (thenav56) [inline apps/export/analyses/excel_exporter.py]:
```suggestion
```

File: apps/export/serializers.py [hunk 10]
```diff
@@ -284,6 +288,15 @@ def validate_report_exporting_widgets(self, widget_ids):
...
     # TODO: def validate_report_structure(self, widget_ids):
 
+    def validate_analysis(self, analysis):
+        if analysis:
+            analysis_object = get_object_or_404(Analysis, id=analysis.id)
+            if analysis_object.project != self.project:
+                raise serializers.ValidationError(
+                    f'Analysis project {analysis.project_id} doesn\'t match current project {self.project.id}'
+                )
+        return analysis
```
Comment (thenav56) [inline apps/export/serializers.py]:
```suggestion
        if analysis=.project != self.project:
            raise serializers.ValidationError(
                f'Analysis project {analysis.project_id} doesn\'t match current project {self.project.id}'
            )
        return analysis
```

File: apps/export/analyses/excel_exporter.py [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+            'Statement',
+            'Entry ID',
+            'Entry',
+            'Entry Link',
+            'Source Link'
+        ]
+
+    def add_analytical_statement_entries(self, analytical_statement_entries):
+        self.analysis_sheet.append([self.titles])
+        for analytical_statement_entry in analytical_statement_entries.iterator():
```
Comment (thenav56) [inline apps/export/analyses/excel_exporter.py]: Add select_related and prefetch_related as required.

File: apps/export/tests/test_mutations.py (hunk 8 above)
Comment (thenav56) [reply apps/export/tests/test_mutations.py:303]: Wow

**Review by thenav56 — CHANGES_REQUESTED** [review]

**Review by thenav56 — APPROVED** [review]

**Pull request merged**